Update security.md

JStrotmann opened the pull request:
The existing security keys should look similar to the example below:
define('AUTH_KEY',         'q/6;#T@S_YwYJNn>d0RzKi<sij3w,!&N;r}-{_jkb[#)%/y%*c&D1CL>m$GW7saq');
define('SECURE_AUTH_KEY',  '75a?T9oLJh+]|-)%l[@<|L;g)d-J]: x_}L@w>Ix=ubE5S!ks3i9Dc_ULLoh!,<F');
define('LOGGED_IN_KEY',    'q{0+_>6dC3Mh-+FX6Uv)Z%zWIVd1@o9W3@&t4pVoCIs;mpiQ|N[kwRQ4J`K8|v|t');
define('NONCE_KEY',        ']iOXSEA|{NSx5]PO:M(B9L%9 aACv!UhmeKUnuE@CiVveuV,y4z:j[-a2-IFk9JH');
define('AUTH_SALT',        'Ty2`+j+Ie|OkB{n5u|DATat}BNT1F71KMlAe|oeH+LfzIqdB)23C4|p7sMxiTft-');
define('SECURE_AUTH_SALT', '?K}>gyV7MlAiNMkr0d|gfO96-S:ySJ)(i)$/BP%NGQ.,>wy|7FZ8zmlZbS+&H>uG');
define('LOGGED_IN_SALT',   'OpWOLAR1+#SKGd{Q]hH3ad^Jl#U_NM5`y+7&*sVt4T1qat%|%eW6.%x_h%qi[Tx0');
define('NONCE_SALT',       'S88+ukoEbM;`+]?X<ge+qS>Wra~% D~(O@eV@wP:pqK8diWBhHLmH+;SLk*[NHTl');

JStrotmann committed:
Update security.md

The existing security keys should look similar to the example below:
define('AUTH_KEY',         'q/6;#T@S_YwYJNn>d0RzKi<sij3w,!&N;r}-{_jkb[#)%/y%*c&D1CL>m$GW7saq');
define('SECURE_AUTH_KEY',  '75a?T9oLJh+]|-)%l[@<|L;g)d-J]: x_}L@w>Ix=ubE5S!ks3i9Dc_ULLoh!,<F');
define('LOGGED_IN_KEY',    'q{0+_>6dC3Mh-+FX6Uv)Z%zWIVd1@o9W3@&t4pVoCIs;mpiQ|N[kwRQ4J`K8|v|t');
define('NONCE_KEY',        ']iOXSEA|{NSx5]PO:M(B9L%9 aACv!UhmeKUnuE@CiVveuV,y4z:j[-a2-IFk9JH');
define('AUTH_SALT',        'Ty2`+j+Ie|OkB{n5u|DATat}BNT1F71KMlAe|oeH+LfzIqdB)23C4|p7sMxiTft-');
define('SECURE_AUTH_SALT', '?K}>gyV7MlAiNMkr0d|gfO96-S:ySJ)(i)$/BP%NGQ.,>wy|7FZ8zmlZbS+&H>uG');
define('LOGGED_IN_SALT',   'OpWOLAR1+#SKGd{Q]hH3ad^Jl#U_NM5`y+7&*sVt4T1qat%|%eW6.%x_h%qi[Tx0');
define('NONCE_SALT',       'S88+ukoEbM;`+]?X<ge+qS>Wra~% D~(O@eV@wP:pqK8diWBhHLmH+;SLk*[NHTl');

JStrotmann committed:
Update security.md

I recommend to add a line about Database prefix while WordPress Database is like a brain for the entire WordPress site and every single information is stored in there thus making it hacker’s favorite target.

ataylorme committed:
review-JStrottmann-security-PR

@JStrotmann I'm suggesting a few formatting changes as well as some change to tone. I'd prefer not to tell hosts what to do, for example enforcing keys, but rather educate them on best practices. Whether or not they force those practices is up to them.

ataylorme commented:
@JStrotmann please review my suggested changes in [this commit](https://github.com/ataylorme/WordPress-Hosting-Best-Practices-Documentation/pull/22/commits/3e295143a77d2e3f689973d732d708f445efb74e) to your pull request.

JStrotmann reviewed (commented):
ack :-)

getsource commented on security/security.md:
```diff
@@ -138,10 +138,17 @@ To backup a database through phpMyAdmin, select the database in phpMyAdmin by cl
 #### Database prefix
 
-WordPress database tables have the same names in all WordPress installations. When the standard wp_ prefix to the database tables' names is used, the whole WordPress database structure is not a secret. Hosters should verify that the prefix to the database tables' names is other than wp_. To secure an existing WordPress installation, turn on the maintenance mode, deactivate all plugins, change the ...
+WordPress database tables have the same names in all WordPress installations. The standard `wp_` database table prefix can be targeted by malicious activity. Consider using a datbase table name prefix other than the default `wp_`. To change an existing WordPress installation follow these steps:
```
I don't think we should be recommending that hosts modify the table prefix for existing installs, but for new installs this makes a lot of sense.

ataylorme replied:
Updated in [32104ac](https://github.com/ataylorme/WordPress-Hosting-Best-Practices-Documentation/pull/22/commits/32104acaad7776125d64fd2fc1e53bbeb82f7e68)

getsource replied:
Looks good!

getsource commented on security/security.md:
```diff
@@ -138,10 +138,17 @@ To backup a database through phpMyAdmin, select the database in phpMyAdmin by cl
+The `wp-config.php` file uses security keys (`AUTH_KEY`, `SECURE_AUTH_KEY`, `LOGGED_IN_KEY`, and `NONCE_KEY`) to ensure better encryption of information stored in the user's cookies. As a best practice security keys should be 60 characters or longer, random and complicated, containing both alphabetic and numeric characters.
 
-**WordPress Security Key Generator https://api.wordpress.org/secret-key/1.1/salt/ can be used to change your security keys.**
+**[The WordPress Security Key Generator](https://api.wordpress.org/secret-key/1.1/salt/) should be used to generate security keys.**
```
Since getting these keys from WP.org makes the install process/security rely on an outside source for keys, is this a good security recommendation?

ataylorme replied:
What do you think about adding a note that `WordPress.org` is a trusted source but folks should be wary about non-trusted sources?

getsource replied:
I think the main issue with the suggestion is that it relies on a single outside source (in this case, WP.org), so if it's compromised, this would be a problem for new installs.

Certainly, using this rather than the default is still better.

ataylorme committed:
Update database prefix recommendation

Recommend using a different database prefix for new installs only and remove instructions for updating an existing installation.
